projects/vdk-plugins/vdk-jobs-troubleshooting/src/vdk/plugin/jobs_troubleshoot/api/troubleshoot_utility.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+from abc import abstractmethod
+
+
+class ITroubleshootUtility:
+    """
+    Allows for troubleshoot utilities to be implemented. It provides a unified
```

Comment:
I would find useful a draft PoC, that gets then refined into an API with implementation accordingly.
The prototype will be implementing an operable troubleshooting implementation

Reply:
I'll add working implementation with the next PR, and it can be used as a reference implementation.